dsc/workflows/base/__init__.py

```diff
@@ -371,119 +327,108 @@ def process_result_messages(self) -> None:
             "ingest_unknown": 0,
         }
 
+        # retrieve and create map of result messages
         sqs_client = SQSClient(
             region=CONFIG.aws_region_name, queue_name=self.output_queue
         )
-
-        for sqs_message in sqs_client.receive():
-            sqs_results_summary["received_messages"] += 1
-
-            message_id = sqs_message["MessageId"]
-            message_body = sqs_message["Body"]
-            receipt_handle = sqs_message["ReceiptHandle"]
-
-            logger.debug(f"Processing result message: {message_id}")
-
+        logger.info(
+            f"Processing DSS result messages from the output queue '{self.output_queue}'"
+        )
+        result_message_map = {}
+        for message in sqs_client.receive():
             try:
-                message_attributes = self._parse_result_message_attrs(
-                    sqs_message["MessageAttributes"]
-                )
-            except InvalidSQSMessageError as exception:
-                logger.error(  # noqa: TRY400
-                    f"Failed to parse 'MessageAttributes' from {message_id}: {exception}"
-                )
-                sqs_results_summary["ingest_unknown"] += 1
-
-                # delete message from the queue
-                sqs_client.delete(
-                    receipt_handle=receipt_handle,
-                    message_id=message_id,
-                )
+                self.validate_result_message(message)
+                item_identifier = message["MessageAttributes"]["PackageID"]["StringValue"]
+                result_message_map[item_identifier] = message
+            except InvalidSQSMessageError:
+                logger.exception(f"Failure parsing message '{message["MessageId"]}'")
```

Comment:
Feeling like maybe this could be a sub-method?

Would be great in this main `finalize()` method to have something like:

```python
result_message_map = self.retrieve_and_parse_result_messages()
```

Reply:
Circling back to this comment after going a little further in code, and I think a `@dataclass` could potentially be a nice fit to represent and encapsulate a result message.

It could provide a few things:
- factory method to parse a "raw" message and return an instance of itself
- internal methods for parsing the JSON of the message Body
- dot notation for common  things like item identifier, `Body.ResultType`, etc.
- etc.

It _might_ even be worth exploring if the `SQSClient()` -- which is a client opinionated for this application -- could yield instances of those.  But that might be a step too far if we just want to keep that client 100% pure  SQS in capabilities.

Reply:
I like the `@dataclass` approach, let me implement that and see how everything looks afterward, @jonavellecuerdo you might hold off until I push those changes

Reply:
Added a `@dataclass`, let me know what you both think!